kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1522 @@
+        IndexType *Acol_indices = rA.index2_data().begin();
+
+        // Filling the index1 vector - DO NOT MAKE PARALLEL THE FOLLOWING LOOP!
+        Arow_indices[0] = 0;
+        for (int i = 0; i < static_cast<int>(rA.size1()); i++)
+            Arow_indices[i + 1] = Arow_indices[i] + indices[i].size();
+
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(rA.size1()); ++i) {
+            const IndexType row_begin = Arow_indices[i];
+            const IndexType row_end = Arow_indices[i + 1];
+            IndexType k = row_begin;
+            for (auto it = indices[i].begin(); it != indices[i].end(); ++it) {
+                Acol_indices[k] = *it;
+                Avalues[k] = 0.0;
+                k++;
+            }
+
+            indices[i].clear(); //deallocating the memory
+
+            std::sort(&Acol_indices[row_begin], &Acol_indices[row_end]);
+        }
+
+        rA.set_filled(indices.size() + 1, nnz);
+
+        Timer::Stop("MatrixStructure");
+    }
+
+    /**
+     * @brief This function is exactly same as the ConstructMatrixStructure() function in base class except that the function has the call to ApplyConstraints function call once the element and conditions compute their equation slave_ids
+     * @param pScheme The pointer to the integration scheme
+     * @param rT The global relation matrix
+     * @param rModelPart The model part to compute
+    */
+    virtual void ConstructRelationMatrixStructure(
+        typename TSchemeType::Pointer pScheme,
+        TSystemMatrixType& rT,
+        ModelPart& rModelPart
+        )
+    {
+        // Filling with zero the matrix (creating the structure)
+        Timer::Start("RelationMatrixStructure");
+
+        std::unordered_map<IndexType, IndexType> solvable_dof_reorder;
+        std::unordered_map<IndexType, IndexSetType> master_indices;
+
+        // Filling with "ones"
+        typedef std::pair<IndexType, IndexType> IndexIndexPairType;
+        typedef std::pair<IndexType, IndexSetType> IndexIndexSetPairType;
+        IndexType counter = 0;
+        for (auto& dof : BaseType::mDofSet) {
+            if (dof.EquationId() < BaseType::mEquationSystemSize) {
+                const IndexType equation_id = dof.EquationId();
+                auto it = mDoFSlaveSet.find(dof);
+                if (it == mDoFSlaveSet.end()) {
+                    solvable_dof_reorder.insert(IndexIndexPairType(equation_id, counter));
+                    master_indices.insert(IndexIndexSetPairType(equation_id, IndexSetType({counter})));
+                    ++counter;
+                } else {
+                    master_indices.insert(IndexIndexSetPairType(equation_id, IndexSetType({})));
+                }
+            }
+        }
+
+        // The process info
+        ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+        /// Definition of the eqautio id vector type
+        EquationIdVectorType ids(3, 0);
```

Comment:
3 because this is done that way in other B&S. The name is only because in this particular case only constraints are computed and then then the name is more descriptive